src/types/zodTypes.ts

```diff
@@ -0,0 +1,172 @@
+import { z } from 'zod';
+import { isDefinedAndNotNullAndNotEmpty } from '@rudderstack/integrations-lib';
+import { isHttpStatusSuccess } from '../v0/util';
+
+export const ProxyMetadataSchema = z.object({
+  jobId: z.number(),
+  attemptNum: z.number(),
+  userId: z.string(),
+  sourceId: z.string(),
+  destinationId: z.string(),
+  workspaceId: z.string(),
+  secret: z.record(z.unknown()),
+  destInfo: z.object({}).optional(),
+  omitempty: z.record(z.unknown()).optional(),
+  dontBatch: z.boolean(),
```

Comment:
Server will sent it by default it could be either true or false. By default in server when this type is initialised it will be `false`
https://github.com/rudderlabs/rudder-server/blob/4fdd81c06fcf13cb03fd08b431106fe109be96b1/router/transformer/transformer.go#L62